lib/Parse/ParseType.cpp

```diff
@@ -925,6 +926,7 @@ ParserResult<TupleTypeRepr> Parser::parseTypeTupleBody() {
     // Parse '= expr' here so we can complain about it directly, rather
     // than dying when we see it.
     if (Tok.is(tok::equal)) {
+      SyntaxParsingContext InitContext(SyntaxContext, SyntaxKind::Initializer);
```

Comment:
If we are using Initializer, can we refactor the part in `FunctionParameter` to initializer too? they are now `equal`+`expr`.

Reply:
Sure thing. I'll do that in follow up PR 👍